WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -193,26 +198,19 @@ def register_require_sign_in
   end
 
   def register
-    @competition = competition_from_params
-    @registration = nil
-    @selected_events ||= []
-    if current_user
-      @registration = Registration.find_by(user_id: current_user.id, competition_id: @competition.id) || @competition.registrations.build(user_id: current_user.id)
-      @selected_events = @registration.saved_and_unsaved_events.empty? ? @registration.user.preferred_events : @registration.saved_and_unsaved_events
-    end
+    set_registration_info!
   end
```

Comment:
To be honest, I think that this method doesn't simplifies things. The code in both actions is not that much similar. Apart from that I can see many possibilities of cleaning the code, please correct me when I'm think in a wrong way.
I'd suggest such a `register` method:
```ruby
def register
  @competition = competition_from_params
  @registration = nil
  @selected_events = [] # ||= was useless here, right?
  if current_user
    @registration = @competition.registrations.find_or_initialize_by(user_id: current_user.id) # Effectively the same, right?
     # We either find or build a registration so there are never nsaved events, right?
    @selected_events = @registration.events.empty? ? @registration.user.preferred_events : @registration.events
  end
end
```